### PR TITLE
Help: New styles for Helpful Articles and search results

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* Re-style Helpful Articles and search results as per mockups
* Add images for the Helpful Articles instead of icons
* Hide everything but the contact information and search results when searching so the page doesn't get so long

**Before**

Screen Shot 2021-01-12 at 3 12 33 PM
Screen Shot 2021-01-12 at 3 12 49 PM
Screen Shot 2021-01-12 at 3 13 06 PM

**After**

Screen Shot 2021-01-12 at 3 10 02 PM
Screen Shot 2021-01-12 at 3 11 40 PM
Screen Shot 2021-01-12 at 3 11 57 PM

#### Testing instructions

* Switch to this PR and navigate to `/help`
* Try a search; search results should appear, the rest of the content should not be visible except for the Contact us block at the bottom.
* Clearing the search field (ESC, hitting the X, or deleting the query) or reloading the page should return the page to its original state.
* Check no results found state
* Make sure there are no other visual bugs or errors. :) 

See #40870

### CODE REVIEW

**Comment by jordesign** [comment]
Looking good to me :) Thanks for the heads up

**Comment by sixhours** [comment]
@sfougnier Hiya! Do you have thoughts on the hover states for the Most Helpful section cards and search results cards? Right now they're too subtle, just using the color scheme's primary color as the text color:

Screen Shot 2021-01-20 at 3 47 21 PM

Top card color is hover state, bottom is un-hovered.

**Comment by ghost** [comment]
Is it possible to maintain this hover from the previous incarnation? 
image

I remember you showing something similar a few weeks back. Maybe if the blue image background conflicts we could tweak it slightly? Not sure how that will work in practice, but it would make the hover state clearer than just showing the text styling change.

**Comment by sixhours** [comment]
Yeah, the hover color matches the image/icon background color, so that looks a bit off to me -- what do you think?

Screen Shot 2021-01-22 at 10 03 24 AM


Maybe we switch the background color for the image/icon to white on hover?

Screen Shot 2021-01-22 at 10 29 31 AM

**Comment by ghost** [comment]
> Yeah, the hover color matches the image/icon background color, so that looks a bit off to me -- what do you think?

Ok, yes. I remember this now. I agree it looks weird when it's the same color.

> Maybe we switch the background color for the image/icon to white on hover?

This is cool! Yeah, this works. One thought, could it be that darker blue (I think something like #3582C4) instead of the white?

**Comment by sixhours** [comment]
Oh I like that!

Screen Shot 2021-01-22 at 11 39 07 AM

**Comment by sixhours** [comment]
👋 @Automattic/serenity This is ready for a code review if y'all can spare a minute. 🙏

**Review by mreishus — COMMENTED** [review]
From a code standpoint, my feedback is that `isSearching` "sounds" like it's a boolean, but it isn't always here; sometimes it's a function that wants a boolean (vs a simple boolean value).

I'd recommend:
* renaming `<HelpSearch>`'s prop `isSearching` -> `onSearch`, 
* renaming `<Help>`'s class attribute function `isSearching = ( status ) => {` -> `setIsSearching = ( status ) => {`, 
*  keeping `<Help>`'s state `isSearching` without changes, since it is a boolean
 

Some examples in case my wording isn't the best, the HelpSearch call in `<Help>` render would look like:
```javascript
<HelpSearch onSearch={this.setIsSearching} />
```

The onSearch attribute function in `<HelpSearch>` would look like:
```javascript
	onSearch = ( searchQuery ) => {
		this.setState( {
			searchQuery,
		} );

		if ( isEmpty( searchQuery ) ) {
			this.props.onSearch( false );
		} else {
			this.props.onSearch( true );
		}

		this.props.recordTracksEvent( 'calypso_help_search', { query: searchQuery } );
	};
```
It's true that `<HelpSearch>` would now have two different `onSearch`es, a class attribute function and a prop, but I think that's ok and generally understood.

These aren't set in stone, just suggestions, would be interested to hear any other opinions.

**Comment by sixhours** [comment]
@mreishus That makes sense, I was wondering if there was a better way. I think your solution is more explicit and easier to grok, too!

File: client/me/help/help-search/index.jsx [hunk 1]
```diff
@@ -84,8 +91,8 @@ export class HelpSearch extends React.PureComponent {
 		}
 
 		const forumBaseUrl = helpLinks.wordpress_forum_links_localized
-			? localizeUrl( 'https://en.forums.wordpress.com/' )
-			: 'https://en.forums.wordpress.com/';
+			? localizeUrl( 'https://wordpress.com/forums' )
```
Comment (kwight) [inline client/me/help/help-search/index.jsx:94]: Will `localizeUrl` give the same results with these updated URLs?

Comment (sixhours) [reply client/me/help/help-search/index.jsx:94]: Yep yep!

Screen Shot 2021-01-26 at 12 10 48 PM

Which then redirects to `https://wordpress.com/fr/forums/?s=hello`

**Comment by sixhours** [comment]
@Automattic/serenity This is ready for another pass when you have a chance! (I know y'all are working against a high priority deadline with other stuff so please let me know if I should ping another group or person in the meantime. :))

**Review by mreishus — APPROVED** [review]
Looks good to me

**Pull request merged**